server/schedulers/balance_leader.go

```diff
@@ -76,6 +76,9 @@ func init() {
 		if err := decoder(conf); err != nil {
 			return nil, err
 		}
+		if conf.Batch == 0 {
```

Comment:
> what will happen if I set the batch to zero when the current batch is 2?

It will keep 2. 
https://github.com/tikv/pd/blob/e278c6c3d83087001843a596834fd2eb080ad281/server/schedulers/balance_leader.go#L102-L105
This change only influences the case when sheduler is created by `ConfigJSONDecoder`